clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -309,11 +309,14 @@ private boolean renameDirectory(Path src, Path dst) throws IOException {
             dstFileStatus = getFileStatus(dst);
             dstExists = true;
             if (!dstFileStatus.isDirectory()) {
-                //TODO (Tals): move all of this to the input validation part of the rename functionality with https://github.com/treeverse/lakeFS/issues/2061
-                // Mimics the behaviour in https://github.com/apache/hadoop/blob/2960d83c255a00a549f8809882cd3b73a6266b6d/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L1527
-                LOG.error("renameDirectory: src {} is a directory and dst {} is a file", src, dst);
-                return false;
+                // Same as https://github.com/apache/hadoop/blob/2960d83c255a00a549f8809882cd3b73a6266b6d/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L1527
+                throw new FileAlreadyExistsException("Failed rename " + src + " to " + dst
+                        + "; source is a directory and dest is a files");
             }
+            // lakefsFs only has non-empty directories. Therefore, if the destination is an existing directory we consider
+            // it to be non-empty. The behaviour is same as https://github.com/apache/hadoop/blob/2960d83c255a00a549f8809882cd3b73a6266b6d/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L1530
+            LOG.error("renameDirectory: failed to rename src {} to dst {}. dst is a non-empty directory.", src, dst);
```

Comment:
This is still with really weird punctuation and rather unlike the error message that I would expect from https://github.com/apache/hadoop/blob/2960d83c255a00a549f8809882cd3b73a6266b6d/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/PathIOException.java#L90.  Think that this should be something like
```suggestion
            LOG.error("renameDirectory: rename src {} to dst {}: non-empty directory.", src, dst);
```

Generally, "failed to..." is an uncommon phrasing for error messages, it's more common to say "WHAT: REASON".

Reply:
missed this I will address this is the logging review I'm planning, sorry about that.

Reply:
fixed in https://github.com/treeverse/lakeFS/pull/2110/files